cni/network/network.go

```diff
@@ -1143,3 +1192,42 @@ func determineSnat() (bool, bool, error) {
 
 	return snatConfig.EnableSnatForDns, snatConfig.EnableSnatOnHost, nil
 }
+
+func convertNnsToCniResult(netRes *nnscontracts.ConfigureContainerNetworkingResponse, ifName string) *cniTypesCurr.Result {
+
+	// This function does not add interfaces to CNI result. Reason being CRI (containerD in baremetal case)
+	// only looks for default interface named "eth0" and this default interface is added in the defer
+	// method of ADD method
+	result := &cniTypesCurr.Result{}
+	var resultIpconfigs []*cniTypesCurr.IPConfig
+
+    if netRes.Interfaces != nil {
+		for i, ni :=  range netRes.Interfaces {
+
+			intIndex := i
+			for _, ip := range ni.Ipaddresses {
+
+				ipWithPrefix := strings.Join([]string { ip.Ip, ip.PrefixLength }, "/")
+				_, ipNet, err := net.ParseCIDR(ipWithPrefix)
+				if err != nil {
+					log.Printf("convertNnsToCniResult error: %s \n", err)
+					continue
+				}
+
+				gateway := net.ParseIP(ip.DefaultGateway)
+				ipConfig := &cniTypesCurr.IPConfig{
+					Address: *ipNet,
+					Gateway: gateway,
+					Version: ip.Version,
+					Interface: &intIndex,
```

Comment:
do we own this struct? pointer to int is interesting

Reply:
This is vendored from container networking repo.